drivers/hwinfo/Kconfig

```diff
@@ -0,0 +1,41 @@
+# Kconfig -HW Info driver configuration options
+
+#
+# Copyright (c) 2019 Alexander Wachter
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config HWINFO
+	bool
+
+menuconfig HWINFO_DEVICE_ID
+	bool "Device id driver"
+	select HWINFO
+	help
+	  Enable device id driver.
+
+if HWINFO_DEVICE_ID
```

Comment:
This looks better but I believe it's still incorrect. This part should be rewritten as 
```
menuconfig HWINFO
	bool "Hardware info drivers"
	help
	  Enable support for hardware info drivers. 

if HWINFO
...
```
Since at these level we are talking exclusively about HWINFO driver. The fact that it provides device ID function is an implementation detail. The code above is consistent with how other driver Kconfig files are written.

Reply:
I agree with @mnkp: AFAIK we're providing a hwinfo driver.  There's not one driver for device id, one for device oui, etc.: they're all things within the hwinfo driver. So
* `HWINFO_DEVICE_ID_NAME` should be `HWINFO_DEVICE_NAME`
* `HWINFO_DEVICE_ID_STM32` should be `HWINFO_STM32`
* `HWINFO_DEVICE_ID_NRF` should be `HWINFO_NRF`

Reply:
We lose the ability to select specific options then. For example if you want the device id you get everything else too even your device does not support the other stuff. Maybe I should change the code as following:
struct hwinfo_device_id_driver_api {
#ifdef HWINFO_DEVICE_ID
		hwinfo_get_device_id_t get_id;
#endif
	};

Reply:
To clarify my point is that if some feature isn't supported by the driver then the api field for it will be null and the wrapper should return `-ENOTSUP`.  We shouldn't try to control via Kconfig which hwinfo capabilities are to be built.

Reply:
@pabigot IMO it would make sense to enable/disable the wrapper and user-space handler.
This saves code space and the user does not have to check for ENOTSUP. I think the probability that more then one or two of the hwinfo's are needed in a build is pretty small. If the API grows the number of unnecessary handlers will increase more and more.

Reply:
I don't approve of not checking for error returns.  MISRA-C doesn't either.  So I don't think that'll fly.

But if you want to make the feature set controllable by Kconfig that's up to you; my opinion is just my opinion and I won't block it either way.

Reply:
@alexanderwachter As I've already suggested it would be very helpful if you added support for reading OUI/MAC id to this API. In this case the differences between enabling the driver and the features provided by the driver would become more clear.

Reply:
@mnkp any suggestion how this API should look like? I don't want to define an API where i have no technical background / idea about how the implementation looks like.
int hwinfo_get_eui48(struct device *dev, u8_t *buffer);
would be my suggestion.

Reply:
I believe that when we discussed it the consensus was to *not* add API for functions that are not currently intended to be implemented, and the only function that had consensus was the id.

Reply:
I'm still not convinced that we properly investigated available options and made an informed decision regarding the implementation. Should the API be extended by adding individual functions, as in the example above. Or should we use an enum.
```
ssize_t hwinfo_get(struct device *dev, enum device_info_type, u8_t *buffer, size_t length);
```